### PR TITLE
core(responsive-images): find offscreen images larger than viewport

### LINKED ISSUE #7236: Lighthouse fails to flag a 24MP image
#### Provide the steps to reproduce
1. Run LH on https://www.portuguesetrails.com/en
2. Observe the horrifying 33 MB of images it loads
2. Look through all the image opportunity audits for their whopping 24 megapixel image https://www.portuguesetrails.com/sites/default/files/gallery/iStock-621859064BicicletasPraia_AG_PT_moedas1.jpg whose dimensions are 6000x4000

#### What is the current behavior?
Lighthouse does not flag this image as being massive.

#### What is the expected behavior?
Lighthouse flags this image.

I'm not really sure what to specifically do about this one though. It falls through the cracks because...

- It passes optimized images because for a 24MP image, it is fairly well compressed at only 2 MB.
- It passes webp images for the same reason.
- It passes "use responsive images" because it is offscreen and has client dimensions of 0.
- It passes "use offscreen" because it is requested after TTI.

I don't really want to change any of those constraints in our existing audits, so it feels like there's just a hole in our coverage for "this image is just way too big".

**Related issues**
#7173

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
Currently there's a loophole that super massive images that are also offscreen can bypass the responsive image check. This PR changes the logic in uses-responsive-images to use 2 vertically stacked viewports (to forgive scrollable images) as the maximum potential size the image could have been displayed at instead of ignoring it completely.

**Related Issues/PRs**
closes #7236

### CODE REVIEW

File: lighthouse-cli/test/smokehouse/test-definitions/byte-efficiency/expectations.js [hunk 1]
```diff
@@ -212,13 +212,14 @@ const expectations = [
           },
         },
         'uses-responsive-images': {
-          displayValue: 'Potential savings of 53\xa0KB',
+          displayValue: 'Potential savings of 82\xa0KB',
```
Comment (connorjclark) [inline lighthouse-cli/test/smokehouse/test-definitions/byte-efficiency/expectations.js]: Can you remove this? It is bad for smoke tests in LR.

Comment (patrickhulce) [reply lighthouse-cli/test/smokehouse/test-definitions/byte-efficiency/expectations.js]: heh, sure I was wondering why it was here anyway :)

File: lighthouse-core/audits/byte-efficiency/uses-responsive-images.js [hunk 2]
```diff
@@ -48,28 +48,42 @@ class UsesResponsiveImages extends ByteEfficiencyAudit {
 
   /**
    * @param {LH.Artifacts.ImageElement} image
-   * @param {number} DPR devicePixelRatio
+   * @param {LH.Artifacts.ViewportDimensions} viewport
    * @return {null|Error|LH.Audit.ByteEfficiencyItem};
    */
-  static computeWaste(image, DPR) {
+  static computeWaste(image, viewport) {
```
Comment (connorjclark) [inline lighthouse-core/audits/byte-efficiency/uses-responsive-images.js]: `ViewportDimensions`? I think we try to keep artifacts named LikeThis in other places, not sure if we do it everywhere tho

Comment (patrickhulce) [reply lighthouse-core/audits/byte-efficiency/uses-responsive-images.js]: sure, we definitely tend to as locals but I'd say it varies a little more once they get passed around. moving in that direction sgtm either way though 👍

**Comment by patrickhulce** [comment]
@exterkamp did you particularly want to review these changes or should I let @connorjclark 's review be sufficient?

File: lighthouse-cli/test/fixtures/byte-efficiency/tester.html [hunk 3]
```diff
@@ -85,15 +85,21 @@ <h2>Byte efficiency tester page</h2>
...
   <!-- PASS(optimized): image is JPEG optimized -->
   <!-- FAIL(webp): image is not WebP optimized -->
-  <!-- FAIL(responsive): image is 25% used at DPR 2 -->
+  <!-- FAIL(responsive): image is 50% used at DPR 3 -->
   <!-- PASS(offscreen): image is onscreen -->
   <img style="width: 256px; height: 170px;" src="lighthouse-1024x680.jpg">
 
+  <!-- PASS(optimized): image is JPEG optimized -->
+  <!-- PASS(webp): image is WebP optimized -->
+  <!-- FAIL(responsive): image is % used at DPR 3 -->
```
Comment (exterkamp) [inline lighthouse-cli/test/fixtures/byte-efficiency/tester.html:94]: Should there be a percent here? Or like <XX% used? Where XX is the threshold?
```html
	  <!-- FAIL(responsive): image is XX% used at DPR 3 -->
```

Comment (patrickhulce) [reply lighthouse-cli/test/fixtures/byte-efficiency/tester.html]: whoops, yes 25% used :)

**Pull request merged**